python/lsst/ap/association/association.py

```diff
@@ -208,15 +211,15 @@ def update_dia_objects(self, dia_objects, updated_obj_ids, exposure):
                 updated_dia_objects.append(dia_object)
             dia_object.set('id', obj_id)
 
-            dia_sources = self.level1_db.load_dia_sources([obj_id])
+            obj_dia_sources = dia_sources[associated_ids == obj_id].copy(deep=True)
```

Comment:
The variable name `obj_dia_sources` is not very clear. I think you're retrieving (copying) all of the DIASources associated with an object ID. But isn't `associated_ids` a whole bunch of object IDs while `obj_id` is a single object ID inside this for-loop? I'm missing something about how this works.

Reply:
Nevermind... this was apparently an outdated version of what you pushed... weird GitHub is weird.